apps/prairielearn/src/middlewares/authn.js

```diff
@@ -65,7 +65,10 @@ module.exports = asyncHandler(async (req, res, next) => {
   }
 
   // Allow auth to be bypassed for local dev mode; also used for tests.
-  // See `pages/authLoginDev` for cookie-based authentication in dev mode.
+  //
+  // To authenticate as specific users in dev mode, set `authType` to some
+  // other value in your config (e.g. "x-auth"). Then, visit the login page
+  // and enter a UID, name, and UIN. This is implemented in `pages/authLogin`.
```

Comment:
Just to clarify, this means that for people doing local course dev they won't see any of these new authn choices? They'll just auto-authn like they've always done?

Reply:
I set `"authType": "x-auth"` but it didn't work as expected for me:
1. Loading PL gave me the unauthenticated homepage, as expected.
2. Clicking "Log in" gave me the login page shown below, as expected.
3. However, clicking "Dev Mode Bypass" or filling in values and clicking "Dev Mode Login" both just sent me back to the unauthenticated homepage.

The local `server.log` didn't have any POSTs but did have some `Unloading cached copies...` lines for reasons that I don't understand!

```
{"assessment_directory":null,"assessment_id":null,"assessment_instance_id":null,"authn_user_id":null,"authn_user_uid":null,"body":{},"course_id":null,"course_instance_id":null,"course_instance_short_name":null,"course_short_name":null,"finished":true,"instance_question_id":null,"ip":"::1","level":"verbose","message":"response","method":"GET","params":{},"path":"/pl/login","question_directory":null,"question_id":null,"response_id":"436d0143-fd95-4cb6-af4f-9f23f0b58149","timestamp":"2023-07-19T23:36:07.546Z","user_id":null,"user_uid":null}
{"assessment_directory":null,"assessment_id":null,"assessment_instance_id":null,"authn_user_id":null,"authn_user_uid":null,"body":{},"course_id":null,"course_instance_id":null,"course_instance_short_name":null,"course_short_name":null,"finished":true,"instance_question_id":null,"ip":"::1","level":"verbose","message":"response","method":"GET","params":{},"path":"/pl/dev_login","question_directory":null,"question_id":null,"response_id":"95676e53-17d2-4aac-acdc-f8765be30f03","timestamp":"2023-07-19T23:36:08.535Z","user_id":null,"user_uid":null}
{"authn_user_id":null,"authn_user_uid":null,"body":{},"ip":"::1","level":"verbose","message":"request","method":"GET","params":{},"path":"/","response_id":"4f297d80-5f09-4d3e-9098-12cfb47063a6","timestamp":"2023-07-19T23:36:08.543Z"}
{"level":"verbose","message":"Unloading cached copies of server.js files in /Users/mwest/git/PrairieLearn/exampleCourse ...","timestamp":"2023-07-19T23:36:08.545Z"}
{"level":"verbose","message":"Successfully unloaded 0 cached files","timestamp":"2023-07-19T23:36:08.545Z"}
{"level":"verbose","message":"Unloading cached copies of server.js files in /Users/mwest/git/PrairieLearn/testCourse ...","timestamp":"2023-07-19T23:36:08.545Z"}
{"level":"verbose","message":"Successfully unloaded 0 cached files","timestamp":"2023-07-19T23:36:08.545Z"}
{"level":"verbose","message":"Unloading cached copies of server.js files in /Users/mwest/git/pl-tam212 ...","timestamp":"2023-07-19T23:36:08.545Z"}
{"level":"verbose","message":"Successfully unloaded 0 cached files","timestamp":"2023-07-19T23:36:08.545Z"}
{"assessment_directory":null,"assessment_id":null,"assessment_instance_id":null,"authn_user_id":null,"authn_user_uid":null,"body":{},"course_id":null,"course_instance_id":null,"course_instance_short_name":null,"course_short_name":null,"finished":true,"instance_question_id":null,"ip":"::1","level":"verbose","message":"response","method":"GET","params":{},"path":"/","question_directory":null,"question_id":null,"response_id":"4f297d80-5f09-4d3e-9098-12cfb47063a6","timestamp":"2023-07-19T23:36:08.550Z","user_id":null,"user_uid":null}
```

<img width="473" alt="image" src="https://github.com/PrairieLearn/PrairieLearn/assets/875833/ea4d3b67-85cf-46d9-a6a2-2d84a1d142fe">

Reply:
This problem appears to have been unique to Safari and the fact that it doesn't store `Secure` cookies for http connections to localhost. #8202 should fix this.